pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Allows to connect to the cloud powered by Pasqal to run sequences."""
+from typing import Dict, List, Mapping, Optional, Union
+
+import sdk
+from numpy.typing import ArrayLike
+
+from pulser import Sequence
+from pulser.devices import Device
+from pulser.register import QubitId
+
+
+class PasqalCloud:
+    """Manager of the connection to the cloud powered by Pasqal.
+
+    The cloud connection enables to run sequences on simulators or on real
+    QPUs.
+
+    Args:
+        client_id: client_id of the API key you are holding for Pasqal
+            cloud.
+        client_secret: client_secret of the API key you are holding for
+            Pasqal cloud.
+        endpoints: Optionally, Pasqal cloud connection URLs.
+        webhook: Optionally, webhook.
+    """
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        endpoints: Optional[sdk.endpoints.Endpoints],
+        webhook: Optional[str] = None,
+    ):
+        """Initializes a connection to the cloud."""
+        self._sdk_connection = sdk.SDK(
+            client_id=client_id,
+            client_secret=client_secret,
+            endpoints=endpoints,
+            webhook=webhook,
+        )
+
+    def create_batch(
```

Comment:
I'm wondering whether we shouldn't validate the parameters specified here. After all, we have the sequence and the variables, so we can just check that `Sequence.build()` runs without errors. Wdyt?

Reply:
Only thing is that it could potentially be confusing for a user why their job passed local validation but not inside the cloud service. On the other hand it would maybe make users happier knowing it in advance a few more times

Reply:
What I mean is that we'll have two stages of validation, with potentially different versions of Pulser. 

Suppose the user creates a sequence `seq`.
They then run this `create_batch` sequence, it fails, and they fix it. I.e. they discover the local validation.
They submit the new fixed sequence to the cloud service (again via `create_batch`). However, now the cloud complains about something else. (which, worst case scenario, is incompatible with local constraints)

But as I said above, this might still be preferred.